vali_objects/utils/subtensor_weight_setter.py

```diff
@@ -66,105 +73,79 @@ def set_weights(self, current_time: int = None):
                 if miner in self.metagraph.hotkeys:
                     challengeperiod_weights.append((
                         self.metagraph.hotkeys.index(miner),
-                        ValiConfig.SET_WEIGHT_MINER_CHALLENGE_PERIOD_WEIGHT
+                        ValiConfig.CHALLENGE_PERIOD_WEIGHT
                     ))
                 else:
                     bt.logging.error(f"Challengeperiod miner {miner} not found in the metagraph.")
 
-            transformed_list = checkpoint_netuid_weights + challengeperiod_weights
+            weights_list = checkpoint_netuid_weights + challengeperiod_weights
+            transformed_hotkeys = [x[0] for x in weights_list]
+
+            metagraph_remaining_hotkeys = list(set(self.metagraph.hotkeys) - set(transformed_hotkeys))
```

Comment:
@jbonilla-tao @suntao99 this is what's driving my request for purge, as we have a number of inactive hotkeys which should be purged before our eliminated miners

Reply:
@sli-tao this is what I was talking about which I think is driving the new challengeperiod logic change

Reply:
This seems redundant if we add purging in the elimination manager

Reply:
sounds good. Updated this to remove and purge any instances of eliminated miners from challenge period, so we don't need to reference it in the main weight setter